presto-testing/src/main/java/io/prestosql/testing/AbstractTestQueryFramework.java

```diff
@@ -59,16 +64,24 @@
 import static io.prestosql.sql.ParsingUtil.createParsingOptions;
 import static io.prestosql.sql.SqlFormatter.formatSql;
 import static io.prestosql.transaction.TransactionBuilder.transaction;
+import static java.lang.Character.MAX_RADIX;
+import static java.lang.Math.abs;
+import static java.lang.Math.min;
 import static java.lang.String.format;
 import static java.util.Collections.emptyList;
 import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.fail;
 
 public abstract class AbstractTestQueryFramework
 {
+    private static final SecureRandom random = new SecureRandom();
+    private static final int RANDOM_SUFFIX_LENGTH = 5;
+    private static final String NAME_PLACEHOLDER = "%NAME%";
+
     private QueryRunner queryRunner;
     private H2QueryRunner h2QueryRunner;
     private SqlParser sqlParser;
+    private final Closer closer = Closer.create();
```

Comment:
can you reuse same closer for multiple methods? Are alrady registered entires cleared after call to `closer.close()`.

Reply:
Yes. Closed things are removed from `closer`. See https://github.com/google/guava/blob/master/guava/src/com/google/common/io/Closer.java#L211

Reply:
It is implementaiton detail, not contract. 
Also the fact how `thrown` field (https://github.com/google/guava/blob/master/guava/src/com/google/common/io/Closer.java#L108) is handles shows that closer cannot be reused.
Please add @BeforeMethod which recreates instance of `Closer`

Reply:
`thrown` here is not used. Also `thrown` is set then closer is going to throw exception on every `close()` and as that it works. What other use case you had in mind.

Also I could reuse `AutoCloseableCloser` here, which we control and does not have `thrown` field.

Reply:
> Also thrown is set then closer is going to throw exception on every close() and as that it works

Yeah exactly.
And we are reusing same `Closer` instance for multiple test methods. Which are totally different execution context. Therefore we should create **new** `Closer` instance for each executed test method. Using `@BeforeMethod` method is an easy way to achieve that.

PS. Yes it would work without it (as we are not using `rethrow` method which makes use of `thrown` field. Yet it is just not in-line with how Closer should be used. How `thrown` filed is used internally very much shows that new execution context, new `Closer` instance should be created.